components/org.wso2.carbon.identity.tenant.resource.manager/src/main/java/org/wso2/carbon/identity/tenant/resource/manager/TenantAwareAxis2ConfigurationContextObserver.java

```
@@ -0,0 +1,188 @@
+        }
+        return eventStreamConfigurationList;
+    }
+
+    private void startTenantFlow(int tenantId, String tenantDomain) {
+
+        PrivilegedCarbonContext.startTenantFlow();
+        PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantId(tenantId);
+        PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(tenantDomain);
+    }
+
+    private void startSuperTenantFlow() {
+
+        PrivilegedCarbonContext.startTenantFlow();
+        PrivilegedCarbonContext carbonContext = PrivilegedCarbonContext.getThreadLocalCarbonContext();
+        carbonContext.setTenantId(MultitenantConstants.SUPER_TENANT_ID);
+        carbonContext.setTenantDomain(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME);
+    }
+
+    private void createTenantEventPublisherConfiguration(int tenantId,
+            List<EventPublisherConfiguration> activeEventPublisherConfigurations) {
+
+        for (EventPublisherConfiguration eventPublisherConfiguration : activeEventPublisherConfigurations) {
+            try {
+                ResourceFile resourceFile = getResourceFile(eventPublisherConfiguration.getEventPublisherName());
+                if (resourceFile != null) {
+                    if (log.isDebugEnabled()) {
+                        log.debug("Configuration store is enabled and there is a file in the configuration store for "
+                                + "the publisher name: " + eventPublisherConfiguration.getEventPublisherName());
+                    }
+                    getResourceManager().addEventPublisherConfiguration(resourceFile);
+                }
+            } catch (ConfigurationManagementException e) {
+                if (e.getErrorCode()
+                        .equals(ConfigurationConstants.ErrorMessages.ERROR_CODE_FEATURE_NOT_ENABLED.getCode())) {
+                    if (log.isDebugEnabled()) {
+                        log.debug("Configuration store is disabled.");
+                    }
+                }
```

This exception is swallowed and only debug info is available on a specific condition. why

---

changed to a warn log and if that case not satisfied exception logged as an error